components/jenkins/storage.tf

```
@@ -0,0 +1,15 @@
+resource "azurerm_storage_account" "jenkinsstg" {
+  name                     = "mgmtstatestore"
+  resource_group_name      = azurerm_resource_group.rg.name
+  location                 = var.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+
+  tags = module.tags.common_tags
+}
+
+resource "azurerm_storage_container" "example" {
+  name                  = "mgmtstatestorecontainer"
```

simpler version, doesn't really need the env bit because we don't combine multiple envs in a subscription but i don't think we would be able to bin that very easily
```suggestion
  name                  = "tfstate-${var.env}"
```